cb_hr_views/models/hr_employee.py

```diff
@@ -0,0 +1,163 @@
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from random import choice
+from string import digits
+
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    def _default_personal_identifier(self):
+        pid = None
+        while not pid or self.env['hr.employee'].search(
+                [('personal_identifier', '=', pid)]):
+            pid = "".join(choice(digits) for i in range(8))
+        return pid
+
+    partner_id = fields.Many2one(
+        'res.partner',
+        required=True,
+        readonly=True,
+        store=True,
+        string='Related partner'
+    )
+    name = fields.Char(
+        compute=False,
+        related='partner_id.name',
+        readonly=True,
+    )
+    firstname = fields.Char(
+        related='partner_id.firstname'
+    )
+    lastname = fields.Char(
+        related='partner_id.lastname'
+    )
+    lastname2 = fields.Char(
+        related='partner_id.lastname2'
+    )
+
+    active = fields.Boolean(
```

Comment:
That is wrong, as active is already related to resource. You should modify the functions toggle_active on partner, resource and hr.employee in order to make it work at the same time.

Reply:
and `res.users` too?